src/WebJobs.Script/Config/AppServiceOptionsSetup.cs

```diff
@@ -20,6 +20,22 @@ public void Configure(AppServiceOptions options)
             options.SubscriptionId = _environment.GetSubscriptionId() ?? string.Empty;
             options.RuntimeSiteName = _environment.GetRuntimeSiteName() ?? string.Empty;
             options.SlotName = _environment.GetSlotName() ?? string.Empty;
+            options.RoleInstance = GetRoleInstance() ?? string.Empty;
+        }
+
+        private string GetRoleInstance()
+        {
+            string instanceName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId);
+            if (string.IsNullOrEmpty(instanceName))
+            {
+                instanceName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ComputerName);
+                if (string.IsNullOrEmpty(instanceName))
+                {
+                    instanceName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ContainerName);
+                }
+            }
+
+            return instanceName;
```

Comment:
@balag0 -- is this still the best way to identify an instance name in Linux? I got this from your PR SDK but wanted to double-check: https://github.com/Azure/azure-webjobs-sdk/pull/2240

Reply:
Yes, In AppService AzureWebsiteInstanceId will be unique per container instance and in consumption ContainerName will be unique.